packages/tab/src/react/list-item.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  names as themeNames,
+  useTheme
+} from '@pluralsight/ps-design-system-theme'
+import { ValueOf } from '@pluralsight/ps-design-system-util'
+import { css } from 'glamor'
+import React, { ForwardRefExoticComponent, RefObject, forwardRef } from 'react'
+
+import stylesheet from '../css'
+
+const styles = {
+  bar: () => css(stylesheet['.psds-tab__list-item__bar']),
+  listItem: (active: boolean, themeName: ValueOf<typeof themeNames>) =>
+    css(
+      stylesheet['.psds-tab__list-item'],
+      stylesheet[`.psds-tab__list-item.psds-theme--${themeName}`],
+      active && stylesheet[`.psds-tab__list-item.psds-tab__list-item--active`],
+      active &&
+        stylesheet[
+          `.psds-tab__list-item.psds-tab__list-item--active.psds-theme--${themeName}`
+        ]
+    ),
+  textInner: () => css(stylesheet['.psds-tab__list-item__text-inner']),
+  textWidth: () => css(stylesheet['.psds-tab__list-item__text'])
+}
+
+export interface BaseListItemProps {
+  id: string | number
+  active?: boolean
+}
+export interface ListItemAnchorProps
+  extends BaseListItemProps,
+    Omit<JSX.IntrinsicElements['a'], 'id' | 'onClick'> {
+  href: string
+  onClick?: (i: number, event: React.MouseEvent<HTMLAnchorElement>) => void
+}
+export interface ListItemButtonProps
+  extends BaseListItemProps,
+    Omit<JSX.IntrinsicElements['button'], 'id' | 'onClick'> {
+  href?: undefined
+  onClick?: (i: number, event: React.MouseEvent<HTMLButtonElement>) => void
+}
+type ListItemElement = HTMLButtonElement | HTMLAnchorElement
+type ListItemProps = ListItemAnchorProps | ListItemButtonProps
+type ListItemComponent = ForwardRefExoticComponent<ListItemProps> & {
+  (props: ListItemAnchorProps, ref?: RefObject<HTMLAnchorElement>): JSX.Element
+  (props: ListItemButtonProps, ref?: RefObject<HTMLButtonElement>): JSX.Element
+}
```

Comment:
Here are the new typings.

1. I did not use the util helpers PropOf and RefFor.

Instead of:

```
PropsOf<'a'>
```

I did:

```
JSX.IntrinsicElements['a']
```

Instead of:

```
RefFor<'a'>
```

I did:

```
RefObject<HTMLAnchorButton>
```

I prefer this approach, because I feel that while the utils may be useful markers for an approach difference, the abstraction isn't super high value.  Removing the abstraction removes a mental translation -- this, every time it's used and every time you inspect the abstraction, which has non-trivial internal typings, to try to figure out what it really means. Once removed, the code is of similar length. And at least in the case of the RefObject, the syntax is more common and familiar.

Thoughts on removing the PropsOf and RefFor abstractions?

Reply:
I can get behind removing the abstractions. I like that they look the same(take a tagName) and they are short. but that's not enough reason to keep them :)